gobblin-runtime/src/main/java/org/apache/gobblin/runtime/local/LocalJobLauncher.java

```diff
@@ -130,20 +134,30 @@ protected void runWorkUnitStream(WorkUnitStream workUnitStream) throws Exception
     Iterator<WorkUnit> workUnitIterator = workUnitStream.getWorkUnits();
     if (!workUnitIterator.hasNext()) {
       LOG.warn("No work units to run");
+      CountEventBuilder countEventBuilder = new CountEventBuilder(JobEvent.WORK_UNITS_EMPTY, 0);
+      this.eventSubmitter.submit(countEventBuilder);
       return;
     }
 
+    this.workUnitCount = 0;
+    while (workUnitIterator.hasNext()) {
```

Comment:
Can you replace the while loop with a for loop? e.g. for (i=workUnitIerator; i.next();) {...}

Further the 2 countEventBuilders can be combined into a single statement at the end of the for loop:
e.g. CountEventBuilder countEventBuilder = (workUnitCount=0)? new CountEventBuilder(JobEvent.WORK_UNITS_EMPTY, 0) : new CountEventBuilder(JobEvent.WORK_UNITS_CREATED, this.workUnitCount);